### PR TITLE
Enable drag'n'drop for pages in unified structure tree

### PR DESCRIPTION
**Pull request by solth**
Enable moving page nodes between structural element nodes in structure tree via drag'n'drop.

**Note**: a bug in the PrimeFaces tree implementation (https://github.com/primefaces/primefaces/issues/3543) prevents the drop index of a page from being determined reliably. At the moment, the drop index is only computed correctly when dropping a node _between_ two other nodes, but not when dropping it directly _on_ the new parent node. The problem will supposedly be fixed when https://github.com/primefaces/primefaces/pull/4809 is merged.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 1]
```diff
@@ -783,34 +786,117 @@ public void onNodeCollapse(NodeCollapseEvent event) {
...
         try {
+            StructureTreeNode dropNode = (StructureTreeNode) dropNodeObject;
+            StructureTreeNode dragNode = (StructureTreeNode) dragNodeObject;
             if (dragNode.getDataObject() instanceof IncludedStructuralElement
                     && dropNode.getDataObject() instanceof IncludedStructuralElement) {
                 checkLogicalDragDrop(dragNode, dropNode);
             } else if (dragNode.getDataObject() instanceof MediaUnit
                     && dropNode.getDataObject() instanceof MediaUnit) {
                 checkPhysicalDragDrop(dragNode, dropNode);
+            } else if (dragNode.getDataObject() instanceof  View && dropNode.getDataObject()
```
Comment (oliver-stoehr) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
            } else if (dragNode.getDataObject() instanceof View
```

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 2]
```diff
@@ -783,34 +786,117 @@ public void onNodeCollapse(NodeCollapseEvent event) {
...
+                    instanceof IncludedStructuralElement) {
```
Comment (oliver-stoehr) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
                     && dropNode.getDataObject() instanceof IncludedStructuralElement) {
```

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 3]
```diff
@@ -783,34 +786,117 @@ public void onNodeCollapse(NodeCollapseEvent event) {
...
+     *
+     * @param fromElement
+     *          IncludedStructuralElement from which View is moved
+     * @param toElement
+     *          IncludedStructuralElement to which View is moved
+     * @param views
+     *          Views to be moved
+     */
+    void moveViews(IncludedStructuralElement fromElement, IncludedStructuralElement toElement, List<View> views) {
+
```
Comment (oliver-stoehr) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
```

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 4]
```diff
@@ -783,34 +786,117 @@ public void onNodeCollapse(NodeCollapseEvent event) {
...
+        toElement.getViews().addAll(views);
+        fromElement.getViews().removeAll(views);
+
```
Comment (oliver-stoehr) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
```

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 5]
```diff
@@ -783,36 +786,117 @@ public void onNodeCollapse(NodeCollapseEvent event) {
...
+     *       bug in the PrimeFaces tree implementation are fixed (https://github.com/primefaces/primefaces/issues/3543)!
+     *
+     * @param fromElement
+     *          IncludedStructuralElement from which View is moved
+     * @param toElement
+     *          IncludedStructuralElement to which View is moved
+     * @param views
+     *          Views to be moved
+     */
+    void moveViews(IncludedStructuralElement fromElement, IncludedStructuralElement toElement, List<View> views) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]: Please add a test

File: Kitodo-API/src/main/java/org/kitodo/api/dataformat/Workpiece.java [hunk 6]
```diff
@@ -177,29 +182,19 @@ public boolean equals(Object obj) {
...
-     */
-    public List<MediaUnit> getAllMediaUnits(String type) {
         List<MediaUnit> mediaUnits = new LinkedList<>(mediaUnit.getChildren());
         for (MediaUnit mediaUnit : mediaUnit.getChildren()) {
             if (Objects.nonNull(mediaUnit)) {
                 mediaUnits = getAllMediaUnitsRecursive(mediaUnit, mediaUnits);
             }
         }
-        return mediaUnits.stream().filter(m -> m.getType().equals(type)).collect(Collectors.toList());
+        mediaUnits.sort(Comparator.comparing(MediaUnit::getOrder));
```
Comment (Kathrin-Huber) [inline Kitodo-API/src/main/java/org/kitodo/api/dataformat/Workpiece.java]: please add a test

**Pull request merged**